### PR TITLE
Adding the OpenJDK support with ability to disable use of ant langtoo…

### PR DESCRIPTION
**Pull request by jlahoda**
…ls build; and specifying a different location for the build.

### CODE REVIEW

**Comment by arvindaprameya** [comment]
+1 approved, Thanks for adding this Jan. Please consider this for 12.0 if that's possible

File: java/java.openjdk.project/src/org/netbeans/modules/java/openjdk/jtreg/ActionProviderImpl.java [hunk 1]
```diff
@@ -409,12 +409,15 @@ private static boolean newStyleXPatch(FileObject testFile) {
...
         Project prj = FileOwnerQuery.getOwner(testFile);
+        Settings settings = prj.getLookup().lookup(Settings.class);
+        boolean useLangtoolsBuild = settings == null || settings.isUseAntBuild();
         List<FileObject> roots = new ArrayList<>();
 
         if (buildDir != null) {
             FileObject repo = prj.getProjectDirectory().getParent().getParent();
             if (repo.getNameExt().equals("langtools") &&
-                ShortcutUtils.getDefault().shouldUseCustomTest(repo.getNameExt(), FileUtil.getRelativePath(repo, testFile))) {
+                ShortcutUtils.getDefault().shouldUseCustomTest(repo.getNameExt(), FileUtil.getRelativePath(repo, testFile)) &&
```
Comment (sdedic) [inline java/java.openjdk.project/src/org/netbeans/modules/java/openjdk/jtreg/ActionProviderImpl.java:419]: Nitpick: maybe a small helper `shouldUseLangtoolsBuild(String repoName, FileObject testFile)` could help to unify this settings lookup + condition: 2 occurrences. Consider when chaning the conditions next time :)

File: java/java.openjdk.project/src/org/netbeans/modules/java/openjdk/project/ActionProviderImpl.java [hunk 2]
```diff
@@ -171,8 +171,14 @@ public void invokeAction(String command, Lookup context) throws IllegalArgumentE
             }
         }
         FileObject scriptFO = script;
+        Settings settings = project.getLookup().lookup(Settings.class);
+        Properties props = new Properties();
+        if (settings.isUseAntBuild()) {
```
Comment (sdedic) [inline java/java.openjdk.project/src/org/netbeans/modules/java/openjdk/project/ActionProviderImpl.java:176]: The other `ActionProviderImpl` checks `settings != null` before dereferencing. But a direct dereference was already there in `COMMAND_BUILD_GENERIC_FAST` branch.

File: java/java.openjdk.project/src/org/netbeans/modules/java/openjdk/project/customizer/BuildCategory.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+                    return ProjectCustomizer.Category.create("build", "Build", null);
+                return null;
+            }
+            @Override
+            public JComponent createComponent(ProjectCustomizer.Category category, Lookup context) {
+                Settings settings = context.lookup(Settings.class);
+                BuildCategory panel = new BuildCategory(settings.isUseAntBuild(), settings.getAntBuildLocation());
+                category.setOkButtonListener(evt -> {
+                    settings.setUseAntBuild(panel.useAntBuild.isSelected());
+                    settings.setJTregLocation(panel.antBuildLocation.getText());
```
Comment (jlahoda) [inline java/java.openjdk.project/src/org/netbeans/modules/java/openjdk/project/customizer/BuildCategory.java:125]: Should use setAntBuildLocation.
```suggestion
                    settings.setAntBuildLocation(panel.antBuildLocation.getText());
```

**Pull request merged**